### PR TITLE
WIP: more descriptions/comments for everything

### PR DESCRIPTION
**Pull request by vbuterin**
Adds descriptions and comments in many more places to make every part of the spec easier to understand and follow.

### CODE REVIEW

File: specs/phase0/beacon-chain.md [hunk 1]
```diff
@@ -241,18 +241,19 @@ The following values are (non-configurable) constants used throughout the specif
...
+| `BASE_REWARD_FACTOR` | `2**6` (= 64) | Global scaling factor for all rewards (and for penalties that are proportional to those rewards). If total deposits are N gwei, then the theoretical max per-epoch reward rate is roughly `BASE_REWARD_FACTOR / sqrt(N)` (eg. 1m ETH deposited -> `10**15` gwei -> `64 / 10**7.5 ~= 0.000002024` per-epoch return -> 0.1663 annual return, as 1 year = `31556926 / 384` epochs)
+| `WHISTLEBLOWER_REWARD_QUOTIENT` | `2**9` (= 512) | If you catch someone and they get slashed, you get `1 / WHISTLEBLOWER_REWARD_QUOTIENT` of their balance. |
+| `PROPOSER_REWARD_QUOTIENT` | `2**3` (= 8) | If you include a message of certain types in your block as a proposer, you get `1 / PROPOSER_REWARD_QUOTIENT` of the reward that they get |
+| `INACTIVITY_PENALTY_QUOTIENT` | `2**25` (= 33,554,432) | If the chain has lost finality for N epochs, then each offline validator loses `N/INACTIVITY_PENALTY_QUOTIENT` of their balance during that epoch. When `N^2/2 = INACTIVITY_PENALTY_QUOTIENT`, the offline validatorsThe goal is to make it unambiguously not profitable to censor >1/3 lose `~1-1/e ~= 63.2%` of their balance. That happens after `2**13` epochs, or 36 days. |
```
Comment (hwwhww) [inline specs/phase0/beacon-chain.md:249]: `INACTIVITY_PENALTY_QUOTIENT` has been fixed to `2**24` on [v012x branch ](https://github.com/ethereum/eth2.0-specs/blob/v012x/specs/phase0/beacon-chain.md#rewards-and-penalties)

File: specs/phase0/beacon-chain.md [hunk 2]
```diff
@@ -181,31 +181,31 @@ The following values are (non-configurable) constants used throughout the specif
...
 
-| Name | Value |
-| - | - |
-| `MIN_DEPOSIT_AMOUNT` | `Gwei(2**0 * 10**9)` (= 1,000,000,000) |
-| `MAX_EFFECTIVE_BALANCE` | `Gwei(2**5 * 10**9)` (= 32,000,000,000) |
-| `EJECTION_BALANCE` | `Gwei(2**4 * 10**9)` (= 16,000,000,000) |
-| `EFFECTIVE_BALANCE_INCREMENT` | `Gwei(2**0 * 10**9)` (= 1,000,000,000) |
+| Name | Value | Description |
+| - | - | - |
+| `MIN_DEPOSIT_AMOUNT` | `Gwei(2**0 * 10**9)` (= 1,000,000,000) | Mininum accepted deposit size |
```
Comment (hwwhww) [inline specs/phase0/beacon-chain.md:205]:
```suggestion
| `MIN_DEPOSIT_AMOUNT` | `Gwei(2**0 * 10**9)` (= 1,000,000,000) | Minimum accepted deposit size |
```

File: specs/phase0/beacon-chain.md [hunk 3]
```diff
@@ -311,6 +320,8 @@ class Checkpoint(Container):
 
 #### `Validator`
 
+All relevant about a validator.
+
```
Comment (protolambda) [inline specs/phase0/beacon-chain.md:324]: Maybe a comment here about the balances being stored separately (for merkleization / change-control purposes) is good? It's does not include "all" about a validator here.

**Comment by protolambda** [comment]
[The annotated spec](https://benjaminion.xyz/eth2-annotated-spec/) by @benjaminion (And credit to many others, @djrtwo in particular) cannot go without mention here. Annotations and insightful comments are great, and make the spec much more readable, and hopefully enables more people to understand and audit it.

The spec has been "minimal" for a long time, often with others reducing the non-code text, but I really do think this PR is a step back in the right direction. However, it raises the question, how far should we go? More text than code?

It's a spec, and even calling them comments is wrong, they could be the majority of the spec. With a note that the code is the decision-maker, since it's not ambiguous unlike text can be. Preserving exact testable definitions through code is great.

cc @JustinDrake You have been the main proponent of minimalism, and although it was valuable to the spec, the text does not have to be like that. What do you think? And can this be a step towards the "Transparent Paper" (if that's how Eth2 as spec document will be called?).

**Comment by JustinDrake** [comment]
> cc @JustinDrake You have been the main proponent of minimalism, and although it was valuable to the spec, the text does not have to be like that. What do you think? And can this be a step towards the "Transparent Paper" (if that's how Eth2 as spec document will be called?).

Agreed that comments should make the spec maximally readable :) I haven't been able to deliver on the transparent paper for 1+ year so I wouldn't hold my breath on it 😂

**Comment by vbuterin** [comment]
Replaced with #1824

**Pull request closed without merging**